nanoprintf.h

```diff
@@ -277,12 +276,10 @@ typedef struct npf_bufputc_ctx {
 } npf_bufputc_ctx_t;
 
 #if NANOPRINTF_USE_LARGE_FORMAT_SPECIFIERS == 1
-  #ifdef _MSC_VER
-    #include <BaseTsd.h>
-    typedef SSIZE_T ssize_t;
-  #else
-    #include <sys/types.h>
-  #endif
+  //compile-time assertion
+  typedef char size_t_and_ptrdiff_t_have_same_size[(sizeof(size_t) == sizeof(ptrdiff_t)) ? 1 : -1];
```

Comment:
similarly, maybe `npf_size_t_and_ptrdiff_t...` just so no poor IDE users start typing `size_t` and get this monstrosity popping up in their autocomplete :)

Reply:
haha good point, I'll make these changes.